src/main/java/org/jahia/modules/sitemap/job/SitemapCreationJob.java

```diff
@@ -51,6 +51,12 @@ public class SitemapCreationJob extends BackgroundJob {
     public void executeJahiaJob(JobExecutionContext jobExecutionContext) throws Exception {
         final ClassLoader initialClassLoader = Thread.currentThread().getContextClassLoader();
         try {
+            // Set the job as running (in case of a scheduled job)
+            JCRTemplate.getInstance().doExecuteWithSystemSession(session ->  {
```

Comment:
Does it mean that the code here https://github.com/Jahia/sitemap/blob/95d5fe5aa881701a31daa4f24a99dc1832a09a02/src/main/java/org/jahia/modules/sitemap/services/impl/SitemapServiceImpl.java#L163 is not necessary anymore?

Reply:
Yes, in case the job is not triggered right away because another one is running (like a request for a translation)